synapse/config/workers.py

```diff
@@ -357,6 +378,23 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
             new_option_name="update_user_directory_from_worker",
         )
 
+        outbound_federation_restricted_to = config.get(
+            "outbound_federation_restricted_to", None
+        )
+        self.outbound_federation_restricted_to = OutboundFederationRestrictedTo(
+            outbound_federation_restricted_to
+        )
+        if outbound_federation_restricted_to:
+            for instance in outbound_federation_restricted_to:
+                if instance != "master" and instance not in self.instance_map:
```

Comment:
Do we need this check now that we're filling out `instance_map` with the main process?

Reply:
I think we can remove the `instance != "master"` part of the check :+1:

I've kept the sanity check to make sure that the `outbound_federation_restricted_to` instance is in the `instance_map` to catch typos and the obvious reconfiguration error.